security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -861,6 +904,77 @@ func (sc *SecretCache) generateFileSecret(connKey ConnKey, token string) (bool,
 	return sdsFromFile, nil, nil
 }
 
+func (sc *SecretCache) generateSecretUsingCaClientWithoutToken(ctx context.Context, connKey ConnKey, t time.Time) (*model.SecretItem, error) {
```

Comment:
This function seems have a lot of duplicate of the following generateSecret(), please try ti refactor and share the common part instead of just copying those common code again.